sdv/multi_table/hma.py

```diff
@@ -431,13 +444,18 @@ def _get_likelihoods(self, table_rows, parent_rows, table_name, foreign_key):
                 A DataFrame of the likelihood of each parent id.
         """
         likelihoods = {}
+
+        data_processor = self._table_synthesizers[table_name]._data_processor
+        table_rows = data_processor.transform(table_rows)
+
         for parent_id, row in parent_rows.iterrows():
             parameters = self._extract_parameters(row, table_name, foreign_key)
             table_meta = self._table_synthesizers[table_name].get_metadata()
             synthesizer = self._synthesizer(table_meta, **self._synthesizer_kwargs)
-            synthesizer.set_parameters(parameters)
+            synthesizer._set_parameters(parameters)
             try:
-                likelihoods[parent_id] = synthesizer.get_likelihood(table_rows)
+                synthesizer._numpy_seed = np.random.get_state()[1]
```

Comment:
why do we need to set the numpy seed here but also set the numpy generator on the method we call on the next line? Shouldn't one be enough?

Reply:
just pass the seed to the method you mean?

Reply:
No I mean we call `synthesizer._get_likelihood` on the next line and inside that method you say 
```python
with np.random.default_rng(self._numpy_seed):
    ....
```
I'm just wondering why we need to do both things and why we don't do them in the same place